airflow/contrib/hooks/spark_submit_hook.py

```diff
@@ -250,8 +250,8 @@ def _build_spark_submit_command(self, application):
             connection_cmd += ["--py-files", self._py_files]
         if self._archives:
             connection_cmd += ["--archives", self._archives]
-        if self._driver_classpath:
-            connection_cmd += ["--driver-classpath", self._driver_classpath]
+        if self._driver_class_path:
+            connection_cmd += ["--driver-class-path", self._driver_class_path]
```

Comment:
Even if there is such a change from `--driver-classpath` to `--driver-class-path`, changing this line into `connection_cmd += ["--driver-class-path", self._driver_classpath]` should be enough, right? I don't think it's necessary to change all other lines.

Reply:
Yes, I understand that but following the convention of other similar parameters(ex: total_executor_cores) I thought of making it consistent.

Reply:
Can you add backward compatibility and a note in the UPDATING.md file? In future versions, we will drop support for the old parameter name.

Reply:
What  version of spark first supports `--driver-class-path`? What version _stopped_ supporting `--driver-classpath`?

Reply:
@ashb This option was introduced in version 1.3 and since then it has been --driver-class-path. There seems to be a typo in airflow while creating spark submit hook.

Reply:
As in there was never `--driver-classpath`? If that's the case then no one could be using this so we don't have to worry about breaking anyone's install :)

Reply:
@mik-laj Backward compatilibility is applicable to some thing that was working before but this option never worked, this seems to be a typo and it is also not a mandatory parameter to pass in the job so nobody cared till now.

Reply:
In this case, I would suggest not to concern about the backward compatibility in this case. Adding a record saying typo fixed in UPDATING would suffice to me.

Reply:
@XD-DENG I have added the record in UPDATING.md. It was closed by mistake. I reopened :)